### DISCUSSION TITLE
Using NavigationBar as RSC results in a webpack error.

### DISCUSSION THREAD

**Discussion by regex5440** [shadcn-ui/ui#3624] [General]
## Description
I am exporting NavigationBar from a component library in mono repo and using it. It results in a webpack error:
>    ![image](https://github.com/shadcn-ui/ui/assets/34965581/804ef681-9b83-4d85-92f8-ca5767f84107)

If I make it a Client component with `"use client";` directive on top of the NavigationBar file then error goes away.

I am confused why is that and if it's known, shad-cn/ui should provide the code with the the directive already.

### Setup
- NextJs + TS with TurboRepo

**Comment by marcioaltoe** [comment]
Exactly same problem and exactly same solution.
